addons/core/functions/server/fnc_serverInit.sqf

```diff
@@ -18,95 +18,129 @@
-} else {
-    VARIABLE_DEFAULT(TFAR_freq_lr_west,true call DFUNC(generateLrSettings));
-    VARIABLE_DEFAULT(TFAR_freq_lr_east,true call DFUNC(generateLrSettings));
-    VARIABLE_DEFAULT(TFAR_freq_lr_independent,true call DFUNC(generateLrSettings));
-};
-
-//Check if all players are running TFAR
-{
-    if(isServer) exitWith {};
-    waitUntil {sleep 0.1;time > 3};
-    if !(isClass(configFile >> "CfgPatches" >> "tfar_core")) exitWith {
-        [player, localize LSTRING(WM_NotLoad)] remoteExec ["globalChat", -2];
-        if (isNull (uiNamespace getVariable ["BIS_fnc_arsenal_cam", objNull])) then {
-            [localize LSTRING(WM_NotLoad_Desc)] call "BIS_fnc_guiMessage";
+GVAR(instanciationIsReady) = false;
+
+// we want to get some of the CBA_settings faster which needs a workaround
+[
+    {
+        missionNamespace getVariable ["CBA_settings", false]
+    },
+    {
+        ["cba_settings_refreshSetting", "TFAR_setting_DefaultRadio_Rifleman_West"] call CBA_fnc_localEvent;
+        ["cba_settings_refreshSetting", "TFAR_setting_DefaultRadio_Rifleman_East"] call CBA_fnc_localEvent;
+        ["cba_settings_refreshSetting", "TFAR_setting_DefaultRadio_Rifleman_Independent"] call CBA_fnc_localEvent;
+    }
+] call CBA_fnc_waitUntilAndExecute;
+
+// after the default radios are set, it's save to instanciate the radios
+[
+    {!(
+        (isNil "TFAR_DefaultRadio_Rifleman_West") && {isNil "TFAR_DefaultRadio_Rifleman_East"} && {isNil "TFAR_DefaultRadio_Rifleman_Independent"} &&
+        {isNil "TFAR_givePersonalRadioToRegularSoldier"} && {isNil "TFAR_instantiate_instantiateAtBriefing"}
+    )},
+    {
+        ["TFAR_RadioRequestEvent", {
+            diag_log format["TFAR_RadioRequestEvent %1 %2",_this,diag_tickTime];//#TODO remove
+
+            if !(params [["_radio_request", [], [[]]],"_requestingUnit"]) exitWith {
+                diag_log format ["TFAR - ERROR:47 - Request Content: %1; Requested By: %2", _radio_reqest, _requestingUnit];
+                ["TFAR_RadioRequestResponseEvent", ["ERROR:47"], _requestingUnit] call CBA_fnc_targetEvent;
+            };
+
+            private _response = [_radio_request, side _requestingUnit] call TFAR_fnc_instanciateRadios;
+            diag_log format["Send TFAR_RadioRequestResponseEvent %1 %2",_response,diag_tickTime]; //#TODO remove
+            ["TFAR_RadioRequestResponseEvent", [_response], _requestingUnit] call CBA_fnc_targetEvent;
+
+            /*
+            //Answer request _response is an array of ["baseclass_ID","baseclass_ID"] of all radios requested
+            ["TFAR_RadioRequestResponseEvent", [(_radio_request call TFAR_fnc_instanciateRadios)], _requestingUnit] call CBA_fnc_targetEvent;
+            */
+        }] call CBA_fnc_addEventHandler;
+        GVAR(instanciationIsReady) = true;
+
+        if (TFAR_instantiate_instantiateAtBriefing) then {
+            // instantiate the radio only on server
+            [] call DFUNC(instanciateRadios);
```

Comment:
Could this run at the same time that the player spawns and tries to init his radios?
I'd say it's unlikely as we have a... 5? second delay before the clientside init starts after spawning and this should run in the briefing screen.

Reply:
Also it's optional feature anyway. So if people have problems with that they should just not enable it.

Reply:
The player won't init his radios until the cba-settings are initialized.
In my opinion, this will prevent the scenario, which you are mentioning.